crates/ruff/src/rules/airflow/rules/tasks.rs

```diff
@@ -0,0 +1,100 @@
+use rustpython_parser::ast;
+use rustpython_parser::ast::{Expr, Ranged};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks that the task variable name is the same as the task_id.
+///
+/// ## Why is this bad?
+/// For consistency, the task variable name should be the same as the task_id.
+/// This makes it easier for you and others to understand the code.
+///
+/// ## Example
+/// ```python
+/// incorrect_name = SomeOperator(task_id="my_task")
+/// async def fetch():
+/// ```
+///
+/// Use instead:
+/// ```python
+/// my_task = SomeOperator(task_id="my_task")
+/// ```
+#[violation]
+pub struct TaskVariableNameNotTaskId;
+
+impl Violation for TaskVariableNameNotTaskId {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Task variable name should be the same as the task_id")
+    }
+}
+
+/// AIR001
+pub(crate) fn task_variable_name(
+    checker: &mut Checker,
+    targets: &[Expr],
+    value: &Expr,
+) -> Option<Diagnostic> {
+    // if we have more than one target, we can't do anything
+    if targets.len() != 1 {
+        return None;
+    }
+
+    let target = &targets[0];
+    let Expr::Name(ast::ExprName { id, .. }) = target else {
+        return None;
+    };
+
+    // if the value is not a call, we can't do anything
+    let call = match value {
+        Expr::Call(call) => call,
+        _ => return None,
+    };
+
+    // if the function doesn't come from airflow, we can't do anything
+    let func_name = match call.func.as_name_expr() {
+        Some(name) => name.id.as_str(),
+        None => return None,
+    };
+    let fully_qualified_func_name = match checker.ctx.find_binding(func_name) {
+        Some(call_path) => match call_path.kind.as_from_importation() {
+            Some(from_importation) => &from_importation.full_name,
+            None => return None,
+        },
+        None => return None,
+    };
+
+    if !fully_qualified_func_name.starts_with("airflow.") {
+        return None;
+    }
```

Comment:
in theory this should only be run if the class being called inherits from `airflow.models.baseoperator.BaseOperator`, but I wanted to get something up to at least start a discussion. Also worth noting that I tried `checker.ctx.resolve_call_path` but couldn't get it to work... could certainly be user error though.